controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -0,0 +1,336 @@
+	_, err := c.doCall(
+		func(ctx context.Context) (interface{}, error) {
+			return c.etcdClient.MoveLeader(ctx, transfereeID)
+		})
+
+	if err != nil {
+		return errors.Wrap(err, "failed to move etcd leader")
+	}
+	return nil
+}
+
+// RemoveMember removes a given member.
+func (c *Client) RemoveMember(memberID uint64) error {
+	_, err := c.doCall(
+		func(ctx context.Context) (interface{}, error) {
+			return c.etcdClient.MemberRemove(ctx, memberID)
+		})
+	if err != nil {
+		return errors.Wrap(err, "failed to remove etcd member")
+	}
+	return nil
+}
+
+// UpdateMemberPeerList updates the given member's list of peers.
+func (c *Client) UpdateMemberPeerList(id uint64, peerURLs []string) (*[]Member, error) {
+	resp, err := c.doCall(
+		func(ctx context.Context) (interface{}, error) {
+			return c.etcdClient.MemberUpdate(ctx, id, peerURLs)
+		})
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to update etcd member %v's peer list to %+v", id, peerURLs)
+	}
+
+	memberUpdateResponse := resp.(*clientv3.MemberUpdateResponse)
+
+	members := make([]Member, 0, len(memberUpdateResponse.Members))
+	for _, m := range memberUpdateResponse.Members {
+		members = append(members, pbMemberToMember(m))
+	}
+
+	return &members, nil
+}
+
+// Alarms retrieves all alarms on a cluster.
+func (c *Client) Alarms() (*[]MemberAlarm, error) {
+	resp, err := c.doCall(
+		func(ctx context.Context) (interface{}, error) {
+			return c.etcdClient.AlarmList(ctx)
+		})
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get alarms for etcd cluster")
+	}
+
+	alarmResponse := resp.(*clientv3.AlarmResponse)
+
+	memberAlarms := make([]MemberAlarm, 0, len(alarmResponse.Alarms))
+	for _, a := range alarmResponse.Alarms {
+		memberAlarms = append(memberAlarms, MemberAlarm{
+			MemberID: a.GetMemberID(),
+			Type:     AlarmType(a.GetAlarm()),
+		})
+	}
+
+	return &memberAlarms, nil
+}
+
+func (c *Client) doCall(call func(context.Context) (interface{}, error)) (interface{}, error) {
+	var lastError error
+	var resp interface{}
+	err := wait.ExponentialBackoff(*c.backoffParams, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+		var err error
+		defer cancel()
+		resp, err = call(ctx)
+		if err == nil {
+			return true, nil
+		}
+		lastError = err
+		return false, nil
+	})
+	if err != nil {
+		return nil, lastError
+	}
+	return resp, nil
+}
+
+// Close closes the etcd client.
+func (c *Client) Close() {
+	c.etcdClient.Close()
+}
+
+// BackoffParams sets the parameters for the client
+func BackoffParams(params wait.Backoff) func(*Client) error {
+	return func(c *Client) error {
+		return c.setBackoffParams(params)
+	}
+}
+
+func (c *Client) setBackoffParams(params wait.Backoff) error {
+	c.backoffParams = &params
+	return nil
+}
+
+// Endpoint changes the etcd endpoint for the client
+func Endpoint(addr string) func(*Client) error {
+	return func(c *Client) error {
+		return c.setEndpoint(addr)
```

Comment:
I'm not sure I would have split this out into a separate function, but it's fine to keep it this way if you want.